### PR TITLE
Updating `CFEMSolver` to use InputParameters construction

### PR DESCRIPTION
**Pull request by andrsd**
This updated the `CFEMSolver` so it has same programming model as the `LBSSolver`, i.e.:
- construction via InputParameters
- options set via `SetOptions`
- boundary conditions set via `boundary_conditions` parameters in an `options` block.

Refs #98

### CODE REVIEW

File: test/modules/cfem_diffusion/c_diffusion_2d_1a_linear.lua [hunk 1]
```diff
@@ -35,27 +35,45 @@ w_vol = logvol.RPPLogicalVolume.Create({xmin=-1000.0,xmax=-0.99999, infy=true, i
...
 s_vol = logvol.RPPLogicalVolume.Create({ymin=-1000.0,ymax=-0.99999, infx=true, infz=true})
 
-e_bndry = 0
-w_bndry = 1
-n_bndry = 2
-s_bndry = 3
+e_bndry = "0"
+w_bndry = "1"
+n_bndry = "2"
+s_bndry = "3"
```
Comment (andrsd) [inline test/modules/cfem_diffusion/c_diffusion_2d_1a_linear.lua:41]: These are supposed to be strings, the C++ code requires `std::string`. This worked since the conversion from `1` to `"1"` was done by the lua layer, but it started to fail when I switched to `InputParameters`.

File: lua/modules/cfem_diffusion/lua_cfem_diff_solver.cc
Comment (zhardy-lanl) [inline lua/modules/cfem_diffusion/lua_cfem_diff_solver.cc:1]: We should consider removing the `lua` prefixes from files in the `lua` directories.

**Review by zhardy-lanl — APPROVED** [review]

File: lua/modules/cfem_diffusion/lua_cfem_diff_solver.cc
Comment (ragusa) [reply lua/modules/cfem_diffusion/lua_cfem_diff_solver.cc:1]: I agree with @zhardy-lanl . @andrsd , should either one of us create an issue for this and tag it as V1.0?

Comment (andrsd) [reply lua/modules/cfem_diffusion/lua_cfem_diff_solver.cc:1]: Opened issue #275 for this

**Review by wdhawkins — APPROVED** [review]

**Comment by andrsd** [comment]
I just noticed I forgot to remove `lua_cfem_diff_solver.h` which is no longer needed with these changes. I will update his PR momentarily.

**Review by ragusa — COMMENTED** [review]
I was wondering why the namespace was changed from cfemdiffusion to diffusion. Will dfemdiffusion also be changed to the namespace diffusion and can this create conflict, now that the solver names also not longer have CFEM or DFEM in their names?

**Comment by zhardy-lanl** [comment]
> I was wondering why the namespace was changed from cfemdiffusion to diffusion. Will dfemdiffusion also be changed to the namespace diffusion and can this create conflict, now that the solver names also not longer have CFEM or DFEM in their names?

I believe we decided to go to `diffusion::CFEMSolver` instead of `cfemdiffusion::Solver` awhile back, no? An analog should be done with DFEM diffusion. I think this is a step towards flattening the namespaces out and reducing name clashes.

**Comment by ragusa** [comment]
Thanks. It was not easy to read everything on a phone screen. My previous comment was wrong. Thanks for the clarification.

**Review by ragusa — APPROVED** [review]

**Comment by andrsd** [comment]
> Will `dfem_diffusion` also be changed to the namespace `diffusion`

Yes. I already have code for that. And this will be done for the other diffusion solvers as well to match the lua module/namespace.

> I believe we decided to go to `diffusion::CFEMSolver` instead of `cfemdiffusion::Solver` awhile back, no?

Yes, this is the plan. Have a `diffusion` C++ namespace with `{CFEM|DFEM|XYZ}Solver` and then matching syntax in the lua layer, i.e. `diffusion.CFEMSolver.Create()`, etc.
